### PR TITLE
refactor: Fix typescript errors in mention rendering

### PR DESCRIPTION
**Pull request by thisisamir98**
refactored mentions

### CODE REVIEW

File: test/unit_tests/util/messageRendererSpec.js [hunk 1]
```diff
@@ -304,6 +304,33 @@ describe('renderMessage', () => {
         testCase: 'displays mention inside code block',
         text: 'salut\n```\n@you\n```',
       },
+      {
+        expected: 'salut<pre><code class="lang-@you"></code></pre>',
+        mentions: [{length: 4, startIndex: 10, userId: 'pain-id'}],
+        testCase: 'displays mention inside code block',
```
Comment (comawill) [inline test/unit_tests/util/messageRendererSpec.js]:
```suggestion
        testCase: 'mention does not affect code language',
```

File: test/unit_tests/util/messageRendererSpec.js [hunk 2]
```diff
@@ -304,6 +304,33 @@ describe('renderMessage', () => {
...
+        text: 'salut\n\n```@you\n```',
+      },
+      {
+        expected:
+          '<a href="https://wire.com/#@you" target="_blank" rel="nofollow noopener noreferrer" data-md-link="true" data-uie-name="markdown-link">text</a>',
+        mentions: [{length: 4, startIndex: 25, userId: 'pain-id'}],
+        testCase: 'displays mention inside code block',
```
Comment (comawill) [inline test/unit_tests/util/messageRendererSpec.js]:
```suggestion
        testCase: 'mention does not affect url',
```

File: test/unit_tests/util/messageRendererSpec.js [hunk 3]
```diff
@@ -304,6 +304,33 @@ describe('renderMessage', () => {
...
+        text: '[text](https://wire.com/#@you)',
+      },
+      {
+        expected:
+          '<a href="https://wire.com/#" title="@you" target="_blank" rel="nofollow noopener noreferrer" data-md-link="true" data-uie-name="markdown-link">text</a>',
+        mentions: [{length: 4, startIndex: 27, userId: 'pain-id'}],
+        testCase: 'displays mention inside code block',
```
Comment (comawill) [inline test/unit_tests/util/messageRendererSpec.js]:
```suggestion
        testCase: 'mention in title is displayed correctly',
```

File: test/unit_tests/util/messageRendererSpec.js [hunk 4]
```diff
@@ -304,6 +304,33 @@ describe('renderMessage', () => {
...
+        text: '[text](https://wire.com/# "@you")',
+      },
+      {
+        expected:
+          '<a href="https://wire.com" target="_blank" rel="nofollow noopener noreferrer" data-md-link="true" data-uie-name="markdown-link"><div class="message-mention" data-uie-name="label-other-mention" data-user-id="pain-id" role="button"><span class="mention-at-sign">@</span>you</div></a>',
+        mentions: [{length: 4, startIndex: 1, userId: 'pain-id'}],
+        testCase: 'displays mention inside code block',
```
Comment (comawill) [inline test/unit_tests/util/messageRendererSpec.js]:
```suggestion
        testCase: 'mention works correctly as part of a link text',
```

File: src/script/util/messageRenderer.ts [hunk 5]
```diff
@@ -170,30 +194,13 @@ export const renderMessage = (message: string, selfId: QualifiedId | null, menti
...
-      const originalToken = originalTokens[index];
-      if (modifiedToken.tag === 'code') {
-        return originalToken;
-      }
-      if (modifiedToken.children) {
-        modifiedToken.children = fixCodeTokens(modifiedToken.children, originalToken.children);
-      }
-      return modifiedToken;
-    });
-  const fixedTokens = fixCodeTokens(modifiedLinksTokens, originalTokens);
```
Comment (comawill) [inline src/script/util/messageRenderer.ts:186]: This code did nothing, but parse the markdown twice

File: src/script/util/messageRenderer.ts [hunk 6]
```diff
@@ -153,7 +177,7 @@ export const renderMessage = (message: string, selfId: QualifiedId | null, menti
     if (!isWireDeepLink && !['autolink', 'linkify'].includes(link.markup)) {
       const title = link.attrGet('title');
       if (title) {
-        link.attrSet('title', title);
+        link.attrSet('title', removeMentionsHashes(title));
       }
       link.attrSet('href', href);
       if (nextToken?.type === 'text') {
```
Comment (comawill) [inline src/script/util/messageRenderer.ts]:
```suggestion
```
remove link.addrSet and the following if (it does nothing), but GitHub does not allow to select all unchanged lines

File: src/script/util/messageRenderer.ts [hunk 7]
```diff
@@ -153,7 +177,7 @@ export const renderMessage = (message: string, selfId: QualifiedId | null, menti
     if (!isWireDeepLink && !['autolink', 'linkify'].includes(link.markup)) {
```
Comment (comawill) [inline src/script/util/messageRenderer.ts:178]:
```suggestion
    link.attrSet('href', href);
    if (!isWireDeepLink && !['autolink', 'linkify'].includes(link.markup)) {
```

File: test/unit_tests/util/messageRendererSpec.js [hunk 8]
```diff
@@ -304,6 +304,33 @@ describe('renderMessage', () => {
...
+        text: '[@you](https://wire.com)',
+      },
     ];
```
Comment (comawill) [inline test/unit_tests/util/messageRendererSpec.js:341]: Please add testcase:
```javascript
  {
        expected:
        '<a href="http://wire.com#@you" target="_blank" rel="nofollow noopener noreferrer">http://wire.com#<div class="message-mention" data-uie-name="label-other-mention" data-user-id="pain-id" role="button"><span class="mention-at-sign">@</span>you</div></a>',
        mentions: [{length: 4, startIndex: 16, userId: 'pain-id'}],
        testCase: 'mentions don\'t affect auto-detected links',
        text: 'http://wire.com#@you',
      },
```

**Review by comawill — CHANGES_REQUESTED** [review]
Found one corner-case, that was not covered yet

**Review by atomrc — APPROVED** [review]

**Review by comawill — APPROVED** [review]

**Pull request merged**